### ISSUE TITLE
eol-last with whitespace breaks everything

### ISSUE THREAD

**Issue by Sahtair** [eslint/eslint#11604]
**Tell us about your environment**

* **ESLint Version:** 5.11.1
* **Node Version:** 10.11.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default with VSCode plugin
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "es6": true,
    "node": true,
    "mocha": true
  },
  "extends": [
    "eslint-config-standard",
    "react-app"
  ],
  "globals": {
    "integrations": true
  },
  "ecmaFeatures": {
    "jsx": true,
    "node": true
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "no-use-before-define": 2,
    "symbol-description": 0
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const init = () => {
   console.log('dsadsa')
  const dsada = 1
}
 
```

**What did you expect to happen?**
I expected it to show me all linter errors.

**What actually happened? Please include the actual, raw output from ESLint.**
It only shows some unrelated linter errors. After removing whitespace from last line it shows all errors

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by Sahtair** [comment]
@platinumazure 
``` js
[Error - 3:59:18 PM] ESLint stack trace:
[Error - 3:59:18 PM] RangeError: Line number out of range (line 6 requested, but only 5 lines present).
    at SourceCode.getIndexFromLoc (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/util/source-code.js:389:19)
    at Object.fix (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/rules/no-multiple-empty-lines.js:116:52)
    at RuleContext.report (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/rule-context.js:127:34)
    at Program:exit.allLines.reduce.concat.reduce (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/rules/no-multiple-empty-lines.js:108:37)
    at Array.reduce (<anonymous>)
    at EventEmitter.Program:exit (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/rules/no-multiple-empty-lines.js:93:22)
    at EventEmitter.emit (events.js:187:15)
    at NodeEventGenerator.applySelector (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/util/node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/util/node-event-generator.js:294:22)
    at NodeEventGenerator.leaveNode (/Users/jurest/Projects/smart-stack/node_modules/eslint/lib/util/node-event-generator.js:317:14)
```

**Comment by platinumazure** [comment]
I'll reopen this (requested info was provided). Sorry for this hiccup.

**Comment by platinumazure** [comment]
I was hoping I or someone else would have a chance to try to recreate the bug before the issue got auto-closed (again). I apologize for not having had a chance to do it myself.

@eslint/eslint-team Could someone please try recreating this bug with the configuration and source code from the original post?
